### ISSUE TITLE
airmon-ng freezes on iwlagn chipset

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#908]
**Reported by breen machine on 17 Jul 2011 15:29 UTC**

The newest version of airmon-ng gets stuck in a while loop around line 352 because the format of the string it gets from dmesg seems to be different than what it expects.

The fix is easy, you just have to stop searching for the word "Link" in the string if you reach the end of it.

My bash scripting skills aren't great so this might have been a stupid way to do it, but here is the relevant section of code with a fix applied:

```
        FOUND="0"
        DONE="0"


        if [ x"$DETECTED_STR" != "x" ]
        then
                while [ $FOUND = "0" ]
                do
                        FOUND=`echo "$DETECTED_STR" | cut -d' ' -f $LINK_POS | grep Link | wc -l`
                        LINK_POS=$(($LINK_POS+1))
                        DONE=`echo "$DETECTED_STR" | cut -d ' ' -f "$LINK_POS"- | wc -c`

                        if [ $DONE = "1" ]
                        then
                                break
                        fi
                done

```

**Comment by aircrack-ng** [comment]
**Comment by breen machine on 17 Jul 2011 15:29 UTC**

Should have added, here is the DETECTED_STR that is causing trouble:
```
[    9.295012] iwlagn 0000:03:00.0: Detected Intel(R) Centrino(R) Ultimate-N 6300 AGN, REV=0x74
```

**Comment by aircrack-ng** [comment]
**Comment by flyingstar16 on 17 Jul 2011 15:29 UTC**

This is really annoying and I've been experiencing this for a while.
Is there a way to get it fixed in the trunk version?

Thanks,
Claudio

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 17 Jul 2011 15:29 UTC**

duplicate with #934

**Comment by aircrack-ng** [comment]
**Comment by misterx on 17 Jul 2011 15:29 UTC**

Fixed in aircrack-ng/aircrack-ng@8cc7803cefd98488dcab0a9c17389a4a18364265.
